### PR TITLE
add Flashing the firmware via cc2538-bsl

### PR DESCRIPTION
**Pull request by der-pw**
In this example I am using a Sonoff dongle and flashing it via cc2538-bsl.

### CODE REVIEW

File: docs/guide/adapters/flashing/flashing_via_cc2538-bsl.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```
+2. Install needed python packages.  
+```bash
+sudo pip3 install pyserial intelhex
+```
+## Download cc2538-bsl
+3. We need the ITead_Sonoff_Zigbee-delay "bootloader" feature.  
+```bash
+mkdir z2m_fw-upgrade
+cd z2m_fw-upgrade
```
Comment (Koenkk) [inline docs/guide/adapters/flashing/flashing_via_cc2538-bsl.md:25]: Doesn't seem to be necessary, the unzip wil already create a separate directory.

**Comment by Koenkk** [comment]
Made some changes, could you check if it is OK?

File: docs/guide/adapters/flashing/flashing_via_cc2538-bsl.md (hunk 1 above)
Comment (der-pw) [reply docs/guide/adapters/flashing/flashing_via_cc2538-bsl.md:25]: Yes, You're right, but I wanted everything in one directory to keep things tidy.
If You want, i can give a hint, for my reasons.

**Comment by der-pw** [comment]
> ... could you check if it is OK?

Yes, it's okay! Thank You!

File: docs/guide/adapters/flashing/flashing_via_cc2538-bsl.md (hunk 1 above)
Comment (Koenkk) [reply docs/guide/adapters/flashing/flashing_via_cc2538-bsl.md:25]: I will rename the directory to cc2538-bsl instead (makes more sense then `z2m_fw-upgrade`)

**Pull request merged**